weave-js/src/components/Panel2/PanelNumber.tsx

```diff
@@ -42,13 +153,17 @@ export const PanelNumber: React.FC<
       }}>
       {nodeValueQuery.result == null
         ? '-'
-        : displayValueNoBarChart(nodeValueQuery.result)}
+        : formatNumber(
+            nodeValueQuery.result,
+            props.config?.propFormat ?? 'Automatic'
```

Comment:
I suspect `propFormat` is untyped. If you add the type to `type PanelNumberProps = Panel2.PanelProps<typeof inputType>;`. like `type PanelNumberProps = Panel2.PanelProps<typeof inputType, {propFormat: string}>;` then you should get the proper typing here